File: Chapter04/Fibonacci.java
import java.io.PrintStream;

public class Fibonacci {

     private int calcFibonacci(int n) {
          if(n <= 1)
              return n;
          else  
              return calcFibonacci(n - 1) + calcFibonacci(n - 2);
     }

     public int calculate(int ctr) {
         int j;
         for (j = 0 ; j < ctr ; j++)
             System.out.print(calcFibonacci(j) + " ");
         System.out.println();
         return 0;
     }

     public Fibonacci() {
     }

     public static void main(String[] argv) {
         if ( argv.length == 0) {
             System.out.println("Usage: fibonacci num-of-sequence-values-to-print");
             System.exit(0);
         } else {
			 Fibonacci test = new Fibonacci();
			test.calculate(new Integer(argv[0]).intValue());
		 }
     }
}

File: Chapter07/fibonacci.c
/* 
 * Simple program to print a certain number of values
 *  in the Fibonacci sequence.
 */

#include <stdio.h>
#include <stdlib.h>

int calc_fib(int n);

int main(int argc, char *argv[]) {
    int i,n;

    if (argc == 2)
      n = atoi(argv[1]);
    else {
      printf("Usage: fibonacci num-of-sequence-values-to-print\n");
      exit(-1);
    }
    for (i=0; i < n; i++)
      printf("%d ", calc_fib(i));
    printf("\n");
    return(0);
}
